File: tyama_atcoderABC005C.rb
#!/usr/bin/ruby
T=gets.to_i
n=gets.to_i
a=gets.split.map(&:to_i)
gets
b=gets.split.map(&:to_i)
#greedy
i=-1
puts b.all?{|e|
	i+=1
	while i<n
		i=n if e-a[i]<0
		break if e-a[i]<=T
		i+=1
	end
	i<n
} ? :yes : :no

File: tyama_atcoderABC005C.cpp
#include <vector>
#include <cstdio>
int main(){
	int T,n,m,e,i;
	scanf("%d%d",&T,&n);
	std::vector<int>a(n);
	for(i=0;i<n;i++)scanf("%d",&a[i]);
	//greedy
	i=-1;
	for(scanf("%d",&m);m;m--){
		scanf("%d",&e);
		for(i++;i<n;i++){
			if(e-a[i]<0)i=n;
			if(e-a[i]<=T)break;
		}
		if(i>=n)break;
	}
	puts(m?"no":"yes");
}